taxi/tests/test_views.py

```diff
@@ -0,0 +1,89 @@
+from http import client
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.forms import CarSearchForm
+from taxi.models import Manufacturer, Car
+from taxi.views import CarListView
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+CAR_LIST_URL = reverse("taxi:car-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="testname",
+            password="test1234",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="Tesla")
+        Manufacturer.objects.create(country="USA")
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(response.status_code, 200)
+        manufacturer = Manufacturer.objects.all()
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturer)
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+
+class PrivateCarTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="testname",
+            password="test1234",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_cars(self):
+        Car.objects.create(model="Tesla", manufacturer=Manufacturer.objects.create(name="Tesla", country="USA"), id=1)
+        Car.objects.create(model="Nissan", manufacturer=Manufacturer.objects.create(name="Nissan", country="Japan"), id=2)
+        response = self.client.get(CAR_LIST_URL)
+        self.assertEqual(response.status_code, 200)
+        cars = Car.objects.all()
+        self.assertEqual(list(response.context["car_list"]), list(cars))
+
+    def test_get_context_data_contains_search_form(self):
+        response = self.client.get(CAR_LIST_URL, {"username": ""})
+        self.assertEqual(response.status_code, 200)
+        self.assertIn("search_form", response.context)
+        search_form = response.context["search_form"]
+        self.assertIsInstance(search_form, CarSearchForm)
+        self.assertEqual(search_form.initial["username"], "")

```

Comment:
The `CarSearchForm` does not have a `username` field according to its definition. Ensure that the correct initial field is being checked in the test.